### PR TITLE
Follow up on Test Plan Review and PROTOTYPE comments for UTF8 String Literals feature.

### PR DESCRIPTION
**Pull request by AlekseyTs**
- Define and implement IUTF8StringOperation node.
- Disallow UTF8StringLiteral conversion for default parameter value.
- Add tests

Test plan https://github.com/dotnet/roslyn/issues/58848

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @RikkiGibson, @dotnet/roslyn-compiler Please review.

File: src/Compilers/Core/Portable/Operations/OperationInterfaces.xml [hunk 1]
```diff
@@ -3273,4 +3273,20 @@
       </Comments>
     </Property>
   </Node>
+  <Node Name="IUTF8StringOperation" Base="IOperation" HasType="true" HasConstantValue="false">
```
Comment (RikkiGibson) [inline src/Compilers/Core/Portable/Operations/OperationInterfaces.xml:3276]: Has this gone through API review?

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/OperationInterfaces.xml (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/OperationInterfaces.xml:3276]: > Has this gone through API review?

Not yet.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler For the second review.

**Review by cston — APPROVED** [review]

**Pull request merged**